File: python/37f_1791B_FollowingDirections.py
def main():
    ans = []
    hmapx = {"L" : -1, "R" : 1, "U" : 0, "D" : 0}
    hmapy = {"L" : 0, "R" : 0, "U" : 1, "D" : -1}
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input().strip()
        x, y, flag = 0, 0, 0
        for i in s:
            x += hmapx[i]
            y += hmapy[i]
            if x == 1 and y == 1:
                flag = 1
                break
        if flag == 0:
            ans.append("NO")
        else:
            ans.append("YES")
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/37f_1791B_FollowingDirections.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, x, y, flag;
    std::string s;
    std::vector<std::string> ans;
    std::unordered_map<char, int> hmapx, hmapy;
    hmapx['L'] = -1, hmapx['R'] = 1, hmapx['U'] = 0, hmapx['D'] = 0;
    hmapy['L'] = 0, hmapy['R'] = 0, hmapy['U'] = 1, hmapy['D'] = -1;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::cin>>s;
        x = 0, y = 0, flag = 0;
        for (auto j : s){
            x += hmapx[j];
            y += hmapy[j];
            if (x == 1 && y == 1){
                flag = 1;
                break;
            }
        }
        if (flag == 0){
            ans.push_back("NO");
        }
        else{
            ans.push_back("YES");
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}